dev-games/tiled/files/tiled-1.11.0-python.patch

```diff
@@ -1,55 +1,38 @@
-diff --git a/qbs/imports/PythonProbe.qbs b/qbs/imports/PythonProbe.qbs
-index 52eae7fc..93b4270c 100644
---- a/qbs/imports/PythonProbe.qbs
-+++ b/qbs/imports/PythonProbe.qbs
-@@ -7,7 +7,6 @@ import qbs.Utilities
- Probe {
-     id: pythonDllProbe
- 
--    property string pythonDir: pythonInstallDir // Input
-     property string buildVariant: qbs.buildVariant // Input
-     property string minVersion: "3.5" // Input
-     property string fileNamePrefix // Output
-@@ -17,18 +16,9 @@ Probe {
-             console.warn(msg + " The Python plugin will not be available.");
-         }
- 
--        if (!pythonDir) {
--            printWarning("PYTHONHOME not set.");
--            return;
--        }
--        if (!File.exists(pythonDir)) {
--            printWarning("The provided Python installation directory '" + pythonDir
--                         + "' does not exist.");
--            return;
--        }
-         var p = new Process();
-         try {
--            var pythonFilePath = FileInfo.joinPaths(pythonDir, "python.exe");
-+            var pythonFilePath = "python";
-             p.exec(pythonFilePath, ["--version"], true);
-             var output = p.readStdOut().trim();
-             var magicPrefix = "Python ";
-@@ -45,9 +35,7 @@ Probe {
-                 return;
-             }
-             if (Utilities.versionCompare(versionNumberString, minVersion) < 0) {
--                printWarning("The Python installation at '" + pythonDir
--                             + "' has version " + versionNumberString + ", but " + minVersion + " or higher "
--                             + "is required.");
-+                printWarning("The Python installation has version " + versionNumberString + ", but " + minVersion + " or higher " + "is required.");
-                 return;
-             }
-             found = true;
+commit addf46e590b0bcf8c0fe60409938b64667c98baf
+Author: Andrei Sabalenka <mechakotik@gmail.com>
+Date:   Wed Aug 21 23:35:47 2024 +0300
+
+    Qbs: Add option to specify custom Python pkg-config name
+    
+    I am Gentoo maintainer of Tiled. In Gentoo, there may be many Python versions installed at the same time for compatibility reasons, and each version has separate name in pkg-config.
+    
+    For example, for Python 3.12 it would be named python-3.12-embed, so running pkg-config for python-embed will fail. Adding this option so it's possible to specify the correct name from build script.
```

Comment:
I'm quite skeptical of this patch because it already does work in Gentoo. It is set up by `python-single-r1_pkg_setup`.

```
$ declare -p PKG_CONFIG_PATH
declare -x PKG_CONFIG_PATH="/var/tmp/portage/dev-games/tiled-1.11.0/temp/python3.12/pkgconfig"
$ ls -l /var/tmp/portage/dev-games/tiled-1.11.0/temp/python3.12/pkgconfig
total 8
lrwxrwxrwx 1 portage portage 41 Aug 25 11:13 python3-embed.pc -> /usr/lib64/pkgconfig/python-3.12-embed.pc
lrwxrwxrwx 1 portage portage 35 Aug 25 11:13 python3.pc -> /usr/lib64/pkgconfig/python-3.12.pc
```

It should be noted that Gentoo fails to provide ptyhon3-embed.pc outside of the portage environment, something that "could" be solved via eselect python. So compiling tiled outside of portage would need this, but compiling it *inside* portage has always behaved correctly as far as I can tell.

Reply:
Just noticed it was working fine without any patches all the time. I didn't even check it because of this misleading warning:
```
WARNING: PYTHONHOME not set. The Python plugin will not be available.
```

Reply:
The split behavior between ordinary compilation and compilation inside portage, is a bit weird to be sure.